src/plugins/external_tool_support/src/ExternalToolManager.cpp

```diff
@@ -343,7 +348,10 @@ void ExternalToolManagerImpl::sl_customToolsLoaded(Task *task) {
     StrStrMap toolPaths;
     foreach (ExternalTool *tool, toolsList) {
         SAFE_POINT(tool, "Tool is NULL", );
-        QString toolPath = addTool(tool);
+        addTool(tool);
```

Comment:
The safe point above does not look right. Can the registry contain nullptrs?

Reply:
The registry isn't supposed to store nullptrs. Actually, the application will crash if you try to add a nullptr to the registry, but it is an inner behavior of the registry and this class shouldn't rely on it.

Reply:
So SAFE_POINT in line 356  should be removed?

Reply:
It's a paranoid style. Let's leave it alone.

Reply:
To me it is not a paranoid style, but a situation when a developer is not sure about the code behavior. 
A developer should always know what to expect from the code he changes and unnecessary SAFE_POINT is the signal of misunderstanding.